### PR TITLE
Feat/update_poetry_in_makefile

### PR DESCRIPTION
**Pull request by sam-deeble**
- Removed duplication with poetry install and update in setup_dev_environment
- Removed poetry shell from setup_dev_environment
- Updated all python usage to use poetry run

### CODE REVIEW

File: Makefile
Comment (adurkan-amido) [inline Makefile:1]: Do these changes mean that we need to amend the docs for running Datastacks? If so, can we include in this PR?

- https://github.com/Ensono/stacks-azure-data/blob/main/docs/workloads/azure/data/etl_pipelines/datastacks.md
- https://github.com/Ensono/stacks-azure-data/blob/main/docs/workloads/azure/data/getting_started/etl_pipelines_deployment_azure.md

File: docs/workloads/azure/data/etl_pipelines/datastacks.md [hunk 1]
```diff
@@ -21,14 +21,13 @@ Datastacks is a utility built to support various functions within the Ensono Sta
 * [Setup project environment](../getting_started/dev_quickstart_data_azure.md)
 
 ```bash
-# Initiate Datastacks using poetry:
-poetry run datastacks
-
-# Run Datastacks from the command line
+# Option 1: Run Datastacks CLI using Poetry's interactive shell (preferred)
```
Comment (adurkan-amido) [inline docs/workloads/azure/data/etl_pipelines/datastacks.md:24]:
```suggestion
# Option 1: Run Datastacks CLI using Poetry's interactive shell (recommended for local development)
```

File: docs/workloads/azure/data/etl_pipelines/datastacks.md [hunk 2]
```diff
@@ -21,14 +21,13 @@ Datastacks is a utility built to support various functions within the Ensono Sta
...
+poetry shell
 datastacks --help
+exit
 
-# Alternatively, run using entrypoint script
-python datastacks/datastacks_cli.py --help
+# Option 2: Run Datastacks CLI using poetry run
```
Comment (adurkan-amido) [inline docs/workloads/azure/data/etl_pipelines/datastacks.md:29]:
```suggestion
# Option 2: Run Datastacks CLI using poetry run (recommended where Poetry shell cannot be used, e.g. CI/CD pipelines)
```

File: docs/workloads/azure/data/etl_pipelines/datastacks.md [hunk 3]
```diff
@@ -86,8 +88,8 @@ In order to generate a new data engineering workload the Datastacks CLI takes a
 Datastacks provides a CLI to conduct data quality checks using the [PySparkle](pysparkle/pysparkle_quickstart.md) library as a backend.
 
 ```bash
-datastacks dq --help
-datastacks dq --config-path "ingest/Ingest_AzureSql_Example/data_quality/ingest_dq.json" --container config
+poetry run datastacks dq --help
+poetry run datastacks dq --config-path "ingest/Ingest_AzureSql_Example/data_quality/ingest_dq.json" --container config
```
Comment (adurkan-amido) [inline docs/workloads/azure/data/etl_pipelines/datastacks.md:92]:
```suggestion
datastacks dq --help
datastacks dq --config-path "ingest/Ingest_AzureSql_Example/data_quality/ingest_dq.json" --container config
```

File: docs/workloads/azure/data/etl_pipelines/datastacks.md [hunk 4]
```diff
@@ -21,14 +21,13 @@ Datastacks is a utility built to support various functions within the Ensono Sta
...
 
 ```bash
-# Initiate Datastacks using poetry:
-poetry run datastacks
-
-# Run Datastacks from the command line
+# Option 1: Run Datastacks CLI using Poetry's interactive shell (preferred)
+poetry shell
 datastacks --help
+exit
```
Comment (adurkan-amido) [inline docs/workloads/azure/data/etl_pipelines/datastacks.md:27]: Any reason we need to have this exit here? Presume people would typically keep the shell open while developing

File: docs/workloads/azure/data/getting_started/etl_pipelines_deployment_azure.md [hunk 5]
```diff
@@ -104,8 +104,8 @@ window_end_default: 2010-01-31
 Use the Datastacks CLI to generate the artifacts for the new workload, using the prepared config file (replacing `path_to_config_file/my_config.yaml` with the appropriate path). Note, a workload with Data Quality steps requires a data platform with a Databricks workspace:
```
Comment (adurkan-amido) [inline docs/workloads/azure/data/getting_started/etl_pipelines_deployment_azure.md:104]:
```suggestion
Use the [Datastacks CLI](../etl_pipelines/datastacks.md#using-the-datastacks-cli) to generate the artifacts for the new workload, using the prepared config file (replacing `path_to_config_file/my_config.yaml` with the appropriate path). Note, a workload with Data Quality steps requires a data platform with a Databricks workspace:
```

File: docs/workloads/azure/data/etl_pipelines/datastacks.md (hunk 4 above)
Comment (sam-deeble) [reply docs/workloads/azure/data/etl_pipelines/datastacks.md:27]: It was to demonstrate where shell isn't required for the second example but the poetry run version will still work if we keep it out (as I have done now)

**Review by adurkan-amido — APPROVED** [review]

**Pull request merged**